base/applications/network/tracert/tracert.cpp

```diff
@@ -0,0 +1,546 @@
+                                     (USHORT)PACKET_SIZE,
+                                     &IpOptionInfo,
+                                     ReplyBuffer,
+                                     ReplySize,
+                                     Info.Timeout);
+            }
+            else
+            {
+                (void)IcmpSendEcho2(Info.hIcmpFile,
+                                     NULL,
+                                     NULL,
+                                     NULL,
+                                     ((PSOCKADDR_IN)Info.Target->ai_addr)->sin_addr.s_addr,
+                                     SendBuffer,
+                                     (USHORT)PACKET_SIZE,
+                                     &IpOptionInfo,
+                                     ReplyBuffer,
+                                     ReplySize,
+                                     Info.Timeout);
+            }
+
+            bool Success;
+            if (Info.Family == AF_INET6)
+            {
+                Success = DecodeV6Response(ReplyBuffer, (Ping == NUM_OF_PINGS), FoundTarget);
+            }
+            else
+            {
+                Success = DecodeV4Response(ReplyBuffer, (Ping == NUM_OF_PINGS), FoundTarget);
+            }
+
+            if (Success == false)
+            {
+                Quit = true;
+                break;
+            }
+
+            if (FoundTarget)
+            {
+                break;
+            }
+        }
+
+        HopCount++;
+        Sleep(100);
+    }
+
+    OutputText(IDS_TRACE_COMPLETE);
+
+    FreeAddrInfoW(Info.Target);
+    if (Info.hIcmpFile)
+    {
+        IcmpCloseHandle(Info.hIcmpFile);
+    }
+
+    return Success;
+}
+
+static bool
+ParseCmdline(int argc, wchar_t *argv[])
+{
+    if (argc < 2)
+    {
+        Usage();
+        return false;
+    }
+
+    for (int i = 1; i < argc; i++)
+    {
+        if (argv[i][0] == '-')
+        {
+            switch (argv[i][1])
+            {
+            case 'd':
+                Info.ResolveAddresses = FALSE;
+                break;
+
+            case 'h':
+                Info.MaxHops = GetULONG(argv[++i]);
+                break;
+
+            case 'j':
+                printf("-j is not yet implemented.\n");
+                return false;
+
+            case 'w':
+                Info.Timeout = GetULONG(argv[++i]);
+                break;
+
+            case '4':
+                Info.Family = AF_INET;
+                break;
+
+            case '6':
+                Info.Family = AF_INET6;
+                break;
+
+            default:
+            {
+                OutputText(IDS_INVALID_OPTION, argv[i]);
+                Usage();
+                return false;
+            }
+            }
+        }
+        else
+        {
+            StringCchCopyW(Info.HostName, NI_MAXHOST, argv[i]);
+            break;
+        }
+    }
+
+    return true;
+}
+
+int wmain(int argc, wchar_t *argv[])
```

Comment:
Add `extern "C"` or `EXTERN_C` to fix the gcc build.

Reply:
Thanks 👍